files/bin/provider-id

```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+
+AVAILABILITY_ZONE=$(imds '/latest/meta-data/placement/availability-zone')
+INSTANCE_ID=$(imds '/latest/meta-data/instance-id')
```

Comment:
We should validate these. I don't know if the script will exit or not if these API calls fail, but we definitely don't want to have bogus provider IDs.

Reply:
You mean validate the format of the instance ID and AZ name? The unit test validates that we use whatever we get from IMDS, and also validates that `provider-id` fails if `imds` does.

Reply:
Ya, I'm just trying to evaluate if there's a possibility that we could get junk from imds or it fails silently (like `INSTANCE_ID` could equal some error message rather than the instance ID).

Reply:
Good point, `imds` wasn't exiting non-zero when `curl` got a response that wasn't 200. Fixed in the latest revision, and added tests for it.